### PR TITLE
remove inline JS to support Content-Security-Policy: default-src 'self'

### PR DESCRIPTION
**Pull request by graingert**
A follow up to #684

### CODE REVIEW

**Comment by graingert** [comment]
@sehmaschine I've finished removing inline JS from the actual Admin. It's still included in the examples and docs though

**Comment by sehmaschine** [comment]
@graingert great stuff. I'll take a look asap ... what do you mean with "examples"?

File: grappelli/templates/admin/prepopulated_fields_js.html [hunk 1]
```diff
@@ -1,25 +1,7 @@
...
-        $(field.id).data('dependency_ids', field['dependency_ids']);
-        $(field.id).data('dependency_list', field['dependency_list']);
-        // prepopulate
-        $(field.id).prepopulate(field['dependency_ids'], field.maxLength);
-    {% endfor %}
-})(grp.jQuery);
+<script type="text/javascript"
+        id="grapelli-prepopulated-fields-js-constants"
+        src="{% static "grapelli/js/old_inlines/prepopulated_field_js.js" %}"
+        data-prepopulated-fields="{% prepopulated_fields_json %}">
```
Comment (graingert) [inline grappelli/templates/admin/prepopulated_fields_js.html]: ah this needs to pass the prepopulated_fields variable

**Comment by graingert** [comment]
Sorry I mean tinymce examples https://github.com/sehmaschine/django-grappelli/tree/master/grappelli/static/grappelli/tinymce/examples

**Comment by sehmaschine** [comment]
sorry, I don't get it ... how are the tinymce examples related to your PR?

**Comment by graingert** [comment]
@sehmaschine I didn't remove the inline js from those files

**Comment by graingert** [comment]
@sehmaschine are there any selenium or manual test scripts for the UI?

**Comment by sehmaschine** [comment]
@graingert I don't think updating the tinymce examples is relevant. There are currently no test scripts for the UI.

**Comment by graingert** [comment]
@sehmaschine the CSP support in django admin landed

**Comment by graingert** [comment]
@sehmaschine what's the status on this?

**Comment by sehmaschine** [comment]
@graingert working on it. this should be part of the upcoming release compatible with django 1.9, right? or did you plan to have these changes with the current grappelli version as well?

**Comment by graingert** [comment]
@sehmaschine this targets 1.9

**Comment by sehmaschine** [comment]
@graingert great. we'll release a new version (2.7.3) today (compatible with Django 1.8) because there's been some minor bugfixes. afterwards, we'll work on compatibility with Django 1.9.

File: grappelli/templates/admin/prepopulated_fields_js.html [hunk 2]
```diff
@@ -1,25 +1,7 @@
...
-        $(field.id).data('dependency_ids', field['dependency_ids']);
-        $(field.id).data('dependency_list', field['dependency_list']);
-        // prepopulate
-        $(field.id).prepopulate(field['dependency_ids'], field.maxLength);
-    {% endfor %}
-})(grp.jQuery);
+<script type="text/javascript"
+        id="grapelli-prepopulated-fields-js-constants"
+        src="{% static "grapelli/js/old_inlines/prepopulated_field_js.js" %}"
+        data-prepopulated-fields="{% prepopulated_fields_json prepopulated_fields %}">
```
Comment (graingert) [inline grappelli/templates/admin/prepopulated_fields_js.html]: this "prepopulated_fields" variable is no longer provided in django 1.10 https://github.com/django/django/pull/5567/files#diff-f0d8070818f4868f26ce6bec567b5bbdR23

**Comment by sehmaschine** [comment]
@graingert is it simple for you to send that PR again (using branch stable/2.8.x)? otherwise I'll try to cherry-pick ...

**Comment by graingert** [comment]
@sehmaschine why not do development like django. Merge into master then cherry-pick to stable

**Comment by sehmaschine** [comment]
@graingert I was not aware that Django does it that way ... we've always tried to keep master stable with grappelli. But I'll give it a try.

**Comment by macher91** [comment]
branch `remove-inline-js-for-csp` was reverted. if I understand there is no option to use CSP with grappelli?

**Pull request merged**